From 44a244b939e79586ebda49b9ab74bc03ec930295 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 10 Oct 2024 14:40:32 +0800
Subject: [PATCH 39/39] fix(util/grub-mkrescue): fix EFI executable name for
 mips64el-efi

Loongson 3A4000/3B4000 firmware expects BOOTMIPS.EFI, not bootmips64el.efi.
---
 util/grub-mkrescue.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/grub-mkrescue.c b/util/grub-mkrescue.c
index 25b861f1f..3759f522d 100644
--- a/util/grub-mkrescue.c
+++ b/util/grub-mkrescue.c
@@ -830,7 +830,7 @@ main (int argc, char *argv[])
       make_image_abs (GRUB_INSTALL_PLATFORM_RISCV64_EFI, "riscv64-efi", imgname);
       free (imgname);
 
-      imgname = grub_util_path_concat (2, efidir_efi_boot, "bootmips64el.efi");
+      imgname = grub_util_path_concat (2, efidir_efi_boot, "BOOTMIPS.EFI");
       make_image_abs (GRUB_INSTALL_PLATFORM_MIPS64EL_EFI, "mips64el-efi", imgname);
       free (imgname);
 
-- 
2.47.0

